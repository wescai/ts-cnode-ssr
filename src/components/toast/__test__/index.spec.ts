import { shallowMount } from "@vue/test-utils";
import Toast from "../toast.vue";
import { toast } from "../";

describe("Toast", () => {
  let test = shallowMount(Toast, { propsData: { message: "test" } });
  it("toast vue template", () => {
    expect(test.props().message).toBe("test");
  });
  it("show", () => {
    let show = toast.show("test");
    expect(show.$el.className).toContain("wu-toast");
    expect(show.$el.textContent).toContain("test");
  });
});
