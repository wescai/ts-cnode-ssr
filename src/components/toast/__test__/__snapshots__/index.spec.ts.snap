// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast loading 1`] = `
<div class="wu-toast wu-toast-center">
  <div class="iconfont icon-loading"></div>
  <div class="wu-toast-content">
    加载中
  </div>
</div>
`;

exports[`Toast loading: 
    <div class="wu-toast wu-toast-center"><div class="iconfont icon-loading"></div> <div class="wu-toast-content">
          加载中
        </div></div> 1`] = `
<div class="wu-toast wu-toast-center">
  <div class="iconfont icon-loading"></div>
  <div class="wu-toast-content">
    加载中
  </div>
</div>
`;

exports[`Toast loading: 1 1`] = `
<div class="wu-toast wu-toast-center">
  <div class="iconfont icon-loading"></div>
  <div class="wu-toast-content">
    加载中
  </div>
</div>
`;

exports[`Toast mount 1`] = `
<div class="wu-toast wu-toast-center">
  <!---->
  <div class="wu-toast-content">
    test
  </div>
</div>
`;

exports[`Toast show 1`] = `
<div class="wu-toast wu-toast-center">
  <!---->
  <div class="wu-toast-content">
    test
  </div>
</div>
`;
